sql/src/main/java/io/crate/statistics/ReservoirSampler.java

```diff
@@ -98,7 +109,45 @@ public Samples getSamples(RelationName relationName, List<Reference> columns, in
         MetaData metaData = clusterService.state().metaData();
         CoordinatorTxnCtx coordinatorTxnCtx = CoordinatorTxnCtx.systemTransactionContext();
         List<Streamer> streamers = Arrays.asList(Symbols.streamerArray(columns));
-        Samples samples = Samples.EMPTY;
+        List<Engine.Searcher> searchersToRelease = new ArrayList<>();
+        RamAccountingContext ramAccounting = new RamAccountingContext(
+            "Reservoir-sampling",
+            circuitBreakerService.getBreaker(CrateCircuitBreakerService.QUERY)
+        );
+        try {
+            return getSamples(
+                columns,
+                maxSamples,
+                docTable,
+                random,
+                metaData,
+                coordinatorTxnCtx,
+                streamers,
+                searchersToRelease,
+                ramAccounting
+            );
+        } finally {
+            ramAccounting.close();
```

Comment:
This de-accounts too early (samples will still be used), but I don't see a better way to release the resources when Samples falls out of scope.

Reply:
Isn't is possible to create the RamAccountingContext in the scope of the analyze job?

Reply:
There is no other job scope on the node where the sampling is running and even if there was, it would probably have the same problem. The caller of `getSamples` immediately creates a response object which then goes over the transport.

We'd need a "response send" callback.